tendrl/gluster_integration/objects/volume/atoms/delete/__init__.py

```diff
@@ -83,13 +86,30 @@ def run(self):
             )
             return False
 
-        NS.etcd_orm.client.delete(
-            "clusters/%s/Volumes/%s" % (
-                NS.tendrl_context.integration_id,
-                self.parameters['Volume.vol_id']
-            ),
-            recursive=True
-        )
+        # Intentionally keeping sleep with high value and blocking
+        # one and not the gevent.sleep(). Make sure no sync etc
+        # happens for this time and then execute etcd delete for volume
+        time.sleep(30)
```

Comment:
Problem here is that the volume gets deleted from gluster cluster but its not deleted from etcd. The gdeploy plugin has returned a success after deletion and then the below etcd delete key was success.

If we used to put a pdb to debug it always used to work, but in normal flow it never used to delete the key and so post check to verify if the volume is deleted successfully used to fail.

So actually waiting inside gdeploy plugin is not the place this is required. Problem is while deletion from etcd.

This explicit time.sleep() helped so kept it like this.